### PR TITLE
feat(e2e testing): add E2E testing module to Terraform

### PR DESCRIPTION
**Pull request by ace-n**
This PR:
1) Adds the E2E testing module to Terraform
2) Adds some sundry changes needed to make the module work

**Notes:**
> ✅  Triggers are successfully created with `setup.sh`
>
> ❌  E2E tests [fail on my project](https://github.com/ace-n/emblem/pull/1/checks?check_run_id=10131402892) _because_ the `EMBLEM_API_URL` env var is not set. _(We can fix this in a forthcoming PR.)_

----
_This PR builds on top of #742._

> _In a future PR, we'll investigate specifying the `content_api_url` using a `google_cloud_run_service` data element:_
> - Using Env/TF variables for this creates an (in @rogerthatdev and I's opinion) **unnecessary** layer of abstraction, which risks being a "premature optimization"

### CODE REVIEW

File: terraform/environments/ops/main.tf [hunk 1]
```diff
@@ -6,3 +6,13 @@ module "emblem_ops" {
   repo_owner      = var.repo_owner
   repo_name       = var.repo_name
 }
+
+module "emblem_website_e2e_test" {
```
Comment (rogerthatdev) [inline terraform/environments/ops/main.tf:10]: let's shorten this to `website_e2e_test`

Comment (ace-n) [reply terraform/environments/ops/main.tf]: Done. 👍

File: terraform/modules/website-e2e-test/iam.tf [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+locals {
+  # Cloud build service account roles
+  pubsub_iam_roles_list = [
+    "roles/cloudscheduler.admin"
+  ]
+}
+
+resource "google_project_iam_member" "web_e2e_iam_members" {
+  project  = var.project_id
+  provider = google
```
Comment (rogerthatdev) [inline terraform/modules/website-e2e-test/iam.tf:12]: ah just realized this is the reason why my testing failed. Ideally we don't want the terraformer service account to grant itself any permissions.  Instead, let's add an entry for this role in the bootstrap step.  All Terraform roles should be added only in the boostrap.

**Review by rogerthatdev — CHANGES_REQUESTED** [review]
I get a failure when testing that only occurs the first time the module is setup. Here's what i did to test this change:

1. Temporarily set the value for count to 0 in the emblem_website_e2e_test module block in `terraform/environments/ops/main.tf`
2.  Completed `setup.sh` with 3 new projects. 
3. Went back and changed the count in step 1 to "1"
4. Set my environment vars and re-ran the terraform using the Cloud Build command used in `configure_delivery.sh`: `gcloud builds submit ./terraform --project="$OPS_PROJECT"     --config=./ops/terraform.cloudbuild.yaml     --substitutions=_ENV="ops",_STATE_GCS_BUCKET_NAME=$STATE_GCS_BUCKET_NAME,_TF_SERVICE_ACCT=$TERRAFORM_SERVICE_ACCOUNT`

The above resulted in an error: 
```sh

│ Error: Error creating Job: googleapi: Error 403: The principal (user or service account) lacks IAM permission "cloudscheduler.jobs.create" for the resource "projects/new-emblem-ops-7b2a/locations/us-central1" (or the resource may not exist).
│ 
│   with module.emblem_website_e2e_test[0].google_cloud_scheduler_job.testing_web_e2e_build_container_job,
│   on ../../modules/website-e2e-test/scheduled-builds.tf line 11, in resource "google_cloud_scheduler_job" "testing_web_e2e_build_container_job":
│   11: resource "google_cloud_scheduler_job" "testing_web_e2e_build_container_job" {

╷
│ Error: Error creating Job: googleapi: Error 403: The principal (user or service account) lacks IAM permission "cloudscheduler.jobs.create" for the resource "projects/new-emblem-ops-7b2a/locations/us-central1" (or the resource may not exist).
│ 
│   with module.emblem_website_e2e_test[0].google_cloud_scheduler_job.testing_web_e2e_run_tests_job,
│   on ../../modules/website-e2e-test/scheduled-builds.tf line 34, in resource "google_cloud_scheduler_job" "testing_web_e2e_run_tests_job":
│   34: resource "google_cloud_scheduler_job" "testing_web_e2e_run_tests_job" {
```


After reviewing further i noticed it's because the cloud scheduler permission is not included in the bootstrap. Let's include there. We should avoid letting terraformer grant itself permissions.  

I've attached a file with the output of the failing apply.
[apply-01.txt](https://github.com/GoogleCloudPlatform/emblem/files/10249929/apply-01.txt)

File: terraform/environments/ops/variables.tf [hunk 3]
```diff
@@ -27,3 +27,8 @@ variable "repo_name" {
   default     = ""
 }
 
+variable "content_api_url" {
+  type        = string
+  description = "The URL of the `staging` project's content API."
+  default     = ""
```
Comment (ace-n) [inline terraform/environments/ops/variables.tf:33]: We _could_ try replacing this with a `data` element _in this PR_.

@rogerthatdev thoughts?

File: terraform/modules/website-e2e-test/variables.tf [hunk 4]
```diff
@@ -23,3 +23,7 @@ variable "content_api_url" {
   type        = string
   description = "A URL pointing to the Content API deployment (staging *or* prod) to test against."
 }
+
+data "google_project" "project" {
```
Comment (rogerthatdev) [inline terraform/modules/website-e2e-test/variables.tf:27]: we won't need this here. `variables.tf` should only include `variable` blocks.

File: terraform/environments/ops/variables.tf (hunk 3 above)
Comment (rogerthatdev) [reply terraform/environments/ops/variables.tf:33]: Let's merge this one like this, and address that next part in the next PR

File: terraform/modules/website-e2e-test/variables.tf (hunk 4 above)
Comment (ace-n) [reply terraform/modules/website-e2e-test/variables.tf]: Removed! _(I think this was purely for the [now-removed] IAM grants. 😄 )_

File: terraform/modules/ops/variables.tf [hunk 5]
```diff
@@ -32,3 +32,9 @@ variable "setup_cd_system" {
   default     = false
   description = "Create deployment triggers. Enable only if Cloud Build has been granted GitHub access."
 }
+
+variable "setup_e2e_tests" {
```
Comment (rogerthatdev) [inline terraform/modules/ops/variables.tf:36]: Sorry I missed this - this variable should be declared only in the `terraform/environments/ops/variables.tf` file

Comment (ace-n) [reply terraform/modules/ops/variables.tf]: Done.

**Review by rogerthatdev — APPROVED** [review]
Fast follow: figure out best way to get the `content-api` url
